core/src/lib.rs

```diff
@@ -373,12 +373,12 @@ pub struct Args {
 
 pub async fn run(
     args: Args,
-) -> (
+) -> Result<(
     impl Future<Output = ()>,
     AppState,
     tracing_appender::non_blocking::WorkerGuard,
     tokio::sync::oneshot::Sender<()>,
-) {
+), Error> {
```

Comment:
I'm a little confused, I don't see any question mark in this function, where could Err(Error) come from?

Reply:
yeah sorry i forgot about, i've updated `run` to actually throw Errors now